src/shared/install.c

```diff
@@ -2367,17 +2372,18 @@ int unit_file_exists(UnitFileScope scope, const LookupPaths *paths, const char *
         return 1;
 }
 
-int unit_file_query_preset(UnitFileScope scope, const char *root_dir, const char *name) {
+static int unit_file_list_presets(UnitFileScope scope, const char *root_dir, Hashmap **presets) {
         _cleanup_strv_free_ char **files = NULL;
         char **p;
         int r;
 
         assert(scope >= 0);
         assert(scope < _UNIT_FILE_SCOPE_MAX);
-        assert(name);
+        assert(presets);
 
-        if (!unit_name_is_valid(name, UNIT_NAME_ANY))
-                return -EINVAL;
+        *presets = hashmap_new(&string_hash_ops);
```

Comment:
The systemd style is to not clobber result pointers in failures, so this should probably use a local variable and store it at the end on success.